doc changes re at-functor and at-layer (#2390)
FluxML/Flux.jl: docs/src/models/functors.md

Flux models are deeply nested structures, and [Functors.jl](https://github.com/FluxML/Functors.jl) provides tools needed to explore such objects, apply functions to the parameters they contain, and re-build them.

!!! compat "Flux ≤ 0.14"

>     All layers used to be defined with the `Functors.@functor` macro.

Reviewer: ```suggestion
    All layers were previously defined with the `Functors.@functor` macro.
```

    This still works, but it is recommended that you use the new [`Flux.@layer`](@ref Flux.@layer) macro instead.

>     Both allow [`params`](@ref Flux.params) to see the parameters inside, and [`gpu`](@ref) to move them to the GPU, but [`Flux.@layer`](@ref Flux.@layer) also overloads printing,

Reviewer: ```suggestion
    Both allow [`Flux.setup`](@ref Flux.setup) to see the parameters inside, and [`gpu`](@ref) to move them to the GPU, but [`Flux.@layer`](@ref Flux.@layer) also overloads printing,
```
